app/src/main/java/com/example/whereto/fragments/PlanTripFragment.java

```diff
@@ -1,21 +1,228 @@
 import com.example.whereto.R;
+import com.google.android.material.slider.LabelFormatter;
+import com.google.android.material.slider.Slider;
+
+import java.text.NumberFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Currency;
 
 public class PlanTripFragment extends Fragment {
-    public PlanTripFragment() {
+    private SwipeDeck cardStack;
+    private ArrayList<Question> questionList;
+    EditText startDate;
+    EditText endDate;
+    DatePickerDialog datePickerDialog;
+
+    public float budget = 0;
+    public float radius = 0;
+    public String tripStart;
+    public String tripEnd;
+    public boolean food = false;
+    public boolean hotels = false;
+    public boolean tours = false;
+    public boolean athletic = false;
+    public boolean arts = false;
+    public boolean shopping = false;
+    public boolean nightlife = false;
+    public boolean beauty = false;
+
+    public PlanTripFragment() {}
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        getQuestionDeck(view);
+        getTripStartDate();
+        getTripEndDate();
+        getBudget();
+
+        Button planTripButton;
+        planTripButton = view.findViewById(R.id.planButton);
+
+        planTripButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                planTrip();
+            }
+        });
     }
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         return inflater.inflate(R.layout.fragment_plan, container, false);
     }
+
+    public void getQuestionDeck(final View view) {
+        questionList = new ArrayList<>();
+        cardStack = (SwipeDeck) view.findViewById(R.id.swipe_deck);
+
+        questionList.add(new Question("food/restaurants"));
+        questionList.add(new Question("hotels"));
+        questionList.add(new Question("tours"));
+        questionList.add(new Question("active/athletic activities"));
+        questionList.add(new Question("arts & entertainment"));
+        questionList.add(new Question("outlet stores/shopping centres/souvenir shops"));
+        questionList.add(new Question("nightlife/bars"));
+        questionList.add(new Question("beauty & spas"));
+
+        final QuestionsAdapter adapter = new QuestionsAdapter(questionList, view.getContext());
+
+        cardStack.setAdapter(adapter);
+
+        cardStack.setEventCallback(new SwipeDeck.SwipeEventCallback() {
+            @Override
+            public void cardSwipedLeft(int position) {}
+
+            @Override
+            public void cardSwipedRight(int position) {
+                String prompt = (questionList.get(position)).getPrompt();
+
+                switch (prompt) {
+                    case "food/restaurants":
```

Comment:
Uh-oh, I see you are using the same strings in line 83ish. We want these to be constants and referencing the same variable. So if there is a typo in one then it doesn't break the system.